src/plugins/data/public/data_sources/datasource/types.ts

```diff
@@ -51,6 +51,7 @@ export interface IDataSourceSettings<T extends IDataSourceMetadata = IDataSource
   id: string;
   type: string;
   name: string;
+  connectionType: 'os' | 'flint';
```

Comment:
could this be a registration point of connection types and then in the query enhancers we can register the conneciton type.

Reply:
yeah that's how it would be done in this implementation, in query enhancers we would specify the connection type during registration. i think that this should be changed though, it would be a redundant field since we could figure this out by seeing if its 'default' or not